src/librbd/mirror/snapshot/CreatePrimaryRequest.cc

```diff
@@ -194,6 +194,13 @@ void CreatePrimaryRequest<I>::unlink_peer() {
       if (info == nullptr) {
         continue;
       }
+      if (info->state == cls::rbd::MIRROR_SNAPSHOT_STATE_PRIMARY_DEMOTED &&
+          m_image_ctx->snap_info.size() > 1 &&
+          info->mirror_peer_uuids.empty()) {
+        peer_uuid = peer;
+        snap_id = snap_it.first;
+        break;
+      }
```

Comment:
Can we please have a comment above the if block about what we are trying to do here. This will be helpful in the future or for a new reader.

Also is this condition something that should be appended to `if (info->mirror_peer_uuids.empty() || !info->complete) {}` because it looks like both the blocks share common logic.

Reply:
I've commented the portion of code to explain rationale. We're also needing to figure out exact specifics of `unlink_peer` and will comment on that. See placeholder containing `TODO:`

Yes those two portions are similar and now there's only one section in the code that addresses that case.